json/src/lib.rs

```diff
@@ -2005,3 +2119,47 @@ where
     Ok(Some(res))
 }
 
+/// deserialize_outpoint deserializes a hex-encoded outpoint
+fn deserialize_outpoint<'de, D>(deserializer: D) -> Result<dashcore::OutPoint, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let str_sequence = String::deserialize(deserializer)?;
+    let str_array: Vec<String> = str_sequence
+        .split('-')
+        .map(|item| item.to_owned())
+        .collect();
+
+    let txid: dashcore::Txid = dashcore::Txid::from_hex(&str_array[0]).unwrap();
+    let vout: u32 = str_array[1].parse().unwrap();
+
+    let outpoint = dashcore::OutPoint{
+        txid: txid,
+        vout: vout,
+    };
+    Ok(outpoint)
+}
+
+/// deserialize_mn_state deserializes a masternode state
+fn deserialize_mn_state<'de, D>(deserializer: D) -> Result<MasternodeState, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let str_sequence = String::deserialize(deserializer)?;
+    
+    Ok(
+        match str_sequence.as_str() {
+        "WAITING_FOR_PROTX" => MasternodeState::MASTERNODE_WAITING_FOR_PROTX,
+        "POSE_BANNED" => MasternodeState::MASTERNODE_POSE_BANNED,
+        "REMOVED" => MasternodeState::MASTERNODE_REMOVED,
+        "OPERATOR_KEY_CHANGED" => MasternodeState::MASTERNODE_OPERATOR_KEY_CHANGED,
+        "PROTX_IP_CHANGED" => MasternodeState::MASTERNODE_PROTX_IP_CHANGED,
+        "READY" => MasternodeState::MASTERNODE_READY,
+        "ERROR" => MasternodeState::MASTERNODE_ERROR,
+        "UNKNOWN" => MasternodeState::UNKNOWN,
+        _ => panic!(),
```

Comment:
Panicking isn't a great idea here, as a new Masternode status could panic all of platform :) Let's create a new State, which would be MasternodeState::NONRECOGNISED .